pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -1473,3 +1473,16 @@ func AddTypeToSecretBinding(secretBinding *gardencorev1beta1.SecretBinding, prov
 	}
 	secretBinding.Provider.Type = strings.Join(types, ",")
 }
+
+// IsCoreDNSClusterProportionalAutoscalingEnabled indicates whether cluster-proportional CoreDNS autoscaling is enabled or not.
+func IsCoreDNSClusterProportionalAutoscalingEnabled(systemComponents *gardencorev1beta1.SystemComponents) bool {
```

Comment:
Does it make sense to make this function more general so that it can be used for other autoscaling modes as well?
```suggestion
func IsCoreDNSAutoscalingUsed(systemComponents *gardencorev1beta1.SystemComponents, autoscalingMode gardencorev1beta1.CoreDNSAutoscalingMode) bool {
```

Reply:
Will be adapted in next commit.